plugins/org.python.pydev/tests_completions/org/python/pydev/ast/codecompletion/revisited/ModulesManagerTest.java

```diff
@@ -300,6 +298,29 @@ public void testGetAllModulesStartingWithRemoval() {
         assertEquals(0, manager.getAllDirectModulesStartingWith("ab").size());
     }
 
+    public void testBuildKeysForRegularEntries() {
+        ProjectModulesManager modulesManager = (ProjectModulesManager) nature2.getAstManager().getModulesManager();
+        String pythonpath = TestDependent.TEST_PYSRC_TESTING_LOC + "\\testWoInit\\";
+        ProjectStub project = new ProjectStub("testProjectStubRefactoring", pythonpath, new IProject[0],
+                new IProject[0]);
+        IProgressMonitor monitor = getProgressMonitor();
+        modulesManager.pythonPathHelper.setPythonPath(pythonpath);
+        ModulesFoundStructure modulesFound = modulesManager.pythonPathHelper.getModulesFoundStructure(project, monitor);
+        PyPublicTreeMap<ModulesKey, ModulesKey> keys = ModulesManager.buildKeysFromModulesFound(monitor, modulesFound);
+
+        ModulesManager.buildKeysForRegularEntries(monitor, modulesFound, keys, false);
+
+        Set<String> expectedKeyNames = new HashSet<String>();
+        expectedKeyNames.add(".__init__"); // testWoInit.__init__
+        expectedKeyNames.add("folder1.__init__");
+        expectedKeyNames.add("folder1.folder2.__init__");
+        expectedKeyNames.add("folder1.folder2.mymod");
+
+        for (ModulesKey key : keys.keySet()) {
```

Comment:
This logic seems backwards for me... what you want to do is check if those entries are in the keys.

Something like:
```
    assertTrue(keys.containsKey(new ModulesKey("folder1.__init__", null)));
    assertFalse(keys.containsKey(new ModulesKey(".__init__", null)));
```